kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaConsumerWrapper.java

```diff
@@ -90,6 +92,19 @@
 
   private boolean waitForReplay = false;
 
+  final List<Future<?>> kafkaConsumreThreads = new ArrayList<>();
+
+  public boolean areKafkaThreadsAreRunning()
+  {
+    for (Future<?> future : kafkaConsumreThreads) {
```

Comment:
Not directly related to this PR, but it does not seem right to use execution service for never ending tasks. It may lead to resource starvation.